### PR TITLE
Update renderer API to support the new Output format

### PR DESCRIPTION
**Pull request by DonJayamanne**
Corresponding PR https://github.com/microsoft/vscode-notebook-renderers/pull/32

### CODE REVIEW

**Review by DavidKutu — APPROVED** [review]

File: src/datascience-ui/ipywidgets/kernel/index.ts [hunk 1]
```diff
@@ -220,6 +221,17 @@ function convertVSCodeOutputToExecutResultOrDisplayData(
...
         }
+    } else {
+        // New API.
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const outputMetadata = request.output.metadata as Record<string, any> | undefined;
+        if (outputMetadata && outputMetadata[request.mimeType] && outputMetadata[request.mimeType].metadata) {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            Object.assign(metadata, outputMetadata[request.mimeType].metadata);
+            if (request.mimeType in outputMetadata[request.mimeType].metadata) {
+                Object.assign(metadata, outputMetadata[request.mimeType].metadata[request.mimeType]);
```
Comment (rchiodo) [inline src/datascience-ui/ipywidgets/kernel/index.ts:232]: This seems weird? 

You assign metadata to the metadata object, and then you reassign it to just the mimetype value? Did you intend to do that?

**Review by rchiodo — CHANGES_REQUESTED** [review]
⏲️

File: src/datascience-ui/ipywidgets/kernel/index.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/datascience-ui/ipywidgets/kernel/index.ts:232]: Yes, the VSC renderer API hasn't been updated, hence not sure what we get here, hence just sending everything (spread everything out & send it all).
I have filed an issue on VSC (and a corresponding tracking issue at our end as well) for them to update their renderer types so we know what we're getting in the request object & what needs to be send to the renderer.

Comment (rchiodo) [reply src/datascience-ui/ipywidgets/kernel/index.ts:232]: Ah okay, part of my confusion was I thought object.assign wholly replaces an object. It just assigns the fields from one object into an already existing object (leaving old fields there).

**Review by rchiodo — APPROVED** [review]

File: src/datascience-ui/ipywidgets/kernel/index.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/datascience-ui/ipywidgets/kernel/index.ts:232]: Very Hacky, just dump everything and hope it works...

**Pull request merged**